### PR TITLE
gadget/install,secboot: test if the tpm can be provisioned

### PR DESCRIPTION
**Pull request by cmatsuoka**
Test if the TPM device can be provisioned before installing the system,
to avoid installing it first and then failing because the TPM is already
provisioned.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

### CODE REVIEW

File: secboot/secboot_tpm.go [hunk 1]
```diff
@@ -54,10 +54,34 @@ var (
...
 )
 
+// CheckTPMProvisionable verifies if the TPM can be provisioned, returning an error
+// if the TPM device does not exist, or if it exists and is already provisioned.
+func CheckTPMProvisionable() error {
+	errPrefix := "cannot check if the TPM can be provisioned"
+
+	tpm, err := sbConnectToDefaultTPM()
+	if err != nil {
+		return fmt.Errorf("%s: %v", errPrefix, err)
```
Comment (zyga) [inline secboot/secboot_tpm.go:69]: Should we use something from golang.org/x/xerrors to wrap `err`? I think we are growing a convention around using it.

Comment (pedronis) [reply secboot/secboot_tpm.go:69]: we aren't growing a convention yet

**Comment by cmatsuoka** [comment]
Blocking this while @chrisccoulson decides if a better API is needed.

**Comment by pedronis** [comment]
this uses concepts and an API that don't exist anymore in secboot, closing

**Pull request closed without merging**